content/2_Bronze/Time_Comp.mdx

```diff
@@ -440,19 +433,45 @@ addition operations take a bit longer than a single addition operation. Another
 example is that although binary search on an array and insertion into an ordered
 set are both $\mathcal{O}(\log n)$, binary search is noticeably faster.
 
-**Constant factor** is entirely ignored in big-O notation. This is fine most of
+**Constant factor** is entirely ignored in Big O notation. This is fine most of
 the time, but if the time limit is particularly tight, you may receive time
 limit exceeded (TLE) with the intended complexity. When this happens, it is
 important to keep the constant factor in mind. For example, a piece of code that
 iterates through all _ordered_ triplets runs in $\mathcal{O}(n^3)$ time might be
 sped up by a factor of $6$ if we only need to iterate through all _unordered_
 triplets.
 
-<!-- For example, if our code currently runs in $\mathcal{O}(n^2)$ time, perhaps we can modify our code to make it run in $\mathcal{O}(n^2/32)$ by using a bitset. (Of course, with big-O notation, $\mathcal{O}(n^2) = O(n^2/32)$.) -->
+<!-- For example, if our code currently runs in $\mathcal{O}(n^2)$ time, perhaps we can modify our code to make it run in $\mathcal{O}(n^2/32)$ by using a bitset. (Of course, with Big O notation, $\mathcal{O}(n^2) = O(n^2/32)$.) -->
 <!-- Bitsets are significantly faster (and so are bitwise operations vs iterating through bits). I don't want to make it sound insignificant -->
 
 For now, don't worry about optimizing constant factors -- just be aware of them.
 
+## Formal Definition of Big O notation
+
+Let $f$ and $g$ be non-negative functions from $\mathbb{R}_{\ge 0}$ to $\mathbb{R}_{\ge 0}$. If there exist
+
+positive constants $n_0$ and $c$ such that $f(n) \le c \cdot g(n)$ whenever $n \ge n_0$, we
+say that $f(n) = \mathcal{O}(g(n))$.
+
+Therefore, it is correct to say that the time complexity of a linear function, $\mathcal{O}(n)$, is $\mathcal{O}(n/2)$, $\mathcal{O}(2n)$, $\mathcal{O}(n^2)$, $\mathcal{O}(2^n)$, $\mathcal{O}(n^n)$, etc.
+
+
+But typically, we write the simplest function out of those that are the most restrictive, thus, $\mathcal{O}(n)$.
+

```

Comment:
```suggestion
```

Reply:
i think this is neede cuz it defines what restrictive and simple means, especially on a beginner lvl.

Reply:
i included it in the previous suggestion